searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchResultViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.hamdan.forzenbook.search.core.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hamdan.forzenbook.core.PostData
+import com.hamdan.forzenbook.search.core.domain.GetPostByStringUseCase
+import com.hamdan.forzenbook.search.core.domain.GetPostByUserIdUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+abstract class BaseSearchResultViewModel(
+    private val getPostByStringUseCase: GetPostByStringUseCase,
+    private val getPostByUserIdUseCase: GetPostByUserIdUseCase
+) : ViewModel() {
+
+    sealed interface SearchResultState {
+        val posts: List<PostData>
+
+        data class Content(override val posts: List<PostData> = listOf()) : SearchResultState
```

Comment:
Use `emptyList()` here for the empty lists you don't intend to use

Reply:
roger